### ISSUE TITLE
Git diffs and IDE/GUI intégration

### ISSUE THREAD

**Issue by stevensmith** [mateodelnorte/meta#226]
## 🤓 Question
Hi 

First off let me just say this project seems awesome and much needed!

I'm trying to convert an existing monorepo into a meta repo and have hit a bit of a wall when it comes to local workflow, just want to make sure I didn't misunderstand or set it up wrong.

I see how you can run a `meta-git status` etc, but from my tests and if I understand this correctly this means it's no longer possible to use a GUI, say vsCode or GitKraken, to see my file diffs before committing, I have to use the `git diff` command exclusively. This is going to be a tough sell for my team I think :(

Did I miss something or am I understanding this correctly? 
Is this a question you've had before? 
Anything that could be done to make this work, like a vsCode plugin for example.

Thanks for everything!
Steve

**Comment by patrickleet** [comment]
open multiple VSCodes?

**Comment by patrickleet** [comment]
if you can't make changes to one project without another, they are probably too tightly coupled, and perhaps shouldn't be separate projects

**Comment by patrickleet** [comment]
multiple VSCodes is a lot easier for using integrated debugger anyway

**Comment by stevensmith** [comment]
Thanks for the reply.

I can open each project in it's own vsCode of course, but then I kind of lose the advantages of the meta repo no?

Here's my use case:

Presently using lerna in a monorepo for web app library to help with npm links and dependency version handling.
Allows me to test changes in shared module locally, then bump versions in consumer modules.

Hoping to do something similar in a meta repo (using meta-npm link meta-bump) to simplify our CI/CD

**Comment by patrickleet** [comment]
> I can open each project in it's own vsCode of course, but then I kind of lose the advantages of the meta repo no?

I don't think so, it's more a logical grouping - the whole point is that there are separate individual pieces that are composed together. They can be built and released separately. Changes should be able to be made independently to each piece. 

substack refers to this as the "idea coverage": https://www.youtube.com/watch?v=faxfLmChjVQ and it's very Unix philosophy-ish.

> then bump versions

I don't fall into the camp that thinks bumping versions is something a developer should care about. That should be automated to take all of the emotion out of it. Is it a fix, feature, or breaking change are the only data needed to calculate new versions.

To do this, I prefer to automate that by using something like `semantic-release` along with conventional commit logs to determine how the version number should change. 

More details here: https://hackernoon.com/bring-in-the-bots-and-let-them-maintain-our-code-gh3s33n9

tl;dr;
Each project gets released once changes land in master which calculates a new version for that module based on the changes described in commit messages. 

It is released. 

A tool like renovate or dependabot notices the new release and makes PRs to other repos that depend on it. Ideally at this point you have good test coverage and reporting and feel confident enabling automerge.

> links

`meta npm link --all` will link all the packages together - this basically just symlinks `node_modules/<whatever>` to the directories.

**Comment by patrickleet** [comment]
There are also vscode workspaces to open all the projects in one VSCode as well, but I've found working with debugger easier with multiple

**Comment by stevensmith** [comment]
Thanks for the extra info. That's definitely the target, we'll try and get it there.

**Comment by patrickleet** [comment]
no problem - feel free to ask more questions here :)
